src/core/dfn-panel.runtime.js

```diff
@@ -32,30 +43,50 @@ function dfnPanel() {
       }
       case "hide": {
         hidePanel(panel);
+        panel = null;
         break;
       }
     }
-  });
+  };
+}
+
+/**
+ * @param {MouseEvent|KeyboardEvent} event
+ */
+function deriveCoordinates(event) {
+  if (event instanceof MouseEvent) {
+    return { x: event.clientX, y: event.clientY };
+  }
+  // Otherwise, KeyboardEvent...
+  const target = /** @type HTMLElement */ (event.target);
+  const rect = target.getBoundingClientRect();
+  // Offset to the middle of the element
+  const x = rect.x + rect.width / 2;
+  // Placed at the bottom of the element
+  const y = rect.y + rect.height;
+  return { x, y };
 }
 
-/** @param {HTMLElement} clickTarget */
-function deriveAction(clickTarget) {
-  const hitALink = !!clickTarget.closest("a");
-  if (clickTarget.closest("dfn, .index-term")) {
-    return hitALink ? null : "show";
+/**
+ * @param {Event} event
```

Comment:
```suggestion
 * @param {MouseEvent|KeyboardEvent} event
```

Reply:
This one only deals with `.target`, which is only relevant for `Event`.